pkg/virt-operator/util/readycheck.go

```diff
@@ -158,8 +162,8 @@ func PodIsUpToDate(pod *k8sv1.Pod, kv *v1.KubeVirt) bool {
 		return false
 	}
 
-	id, ok := pod.Annotations[v1.InstallStrategyIdentifierAnnotation]
-	if !ok || id != kv.Status.TargetDeploymentID {
+	podHash, ok := pod.Annotations[v1.KubeVirtPodTemplateAnnotationHash]
+	if hash != "" && (!ok || podHash != hash) {
```

Comment:
```suggestion
	if !ok || podHash != hash {
```

This should be sufficient?

Reply:
Call `GetPodTemplateHash` here with the already passed in `pod`?